modules/navier_stokes/src/auxkernels/kEpsilonViscosityAux.C

```diff
@@ -166,60 +115,87 @@ kEpsilonViscosityAux::computeValue()
+      mu_wall = rho * Utility::pow<2>(u_tau) * min_wall_dist / parallel_speed;
+      mut_log = mu_wall - mu;
+    }
+    else if (_wall_treatment == "neq")
     {
-      const auto wall_val =
-          _rho(current_argument, state) * u_tau * min_wall_dist /
-              (1 / NS::von_karman_constant * std::log(NS::E_turb_constant * y_plus)) -
-          _mu(current_argument, state);
-      mu_t_wall = wall_val.value();
+      // Assign non-equilibrium wall function value
+      y_plus = min_wall_dist * std::sqrt(std::sqrt(_C_mu) * _k(current_argument, state)) * rho / mu;
+      mu_wall = mu * (NS::von_karman_constant * y_plus /
+                      std::log(std::max(NS::E_turb_constant * y_plus, 1 + 1e-4)));
+      mut_log = mu_wall - mu;
     }
-    else // buffer layer
+    else
+      mooseAssert(false, "Should not reach here");
+
+    if (y_plus <= 5.0)
+      // sub-laminar layer
+      mu_t = 0.0;
+    else if (y_plus >= 30.0)
+      // log-layer
+      mu_t = std::max(mut_log.value(), NS::mu_t_low_limit);
+    else
     {
-      const auto wall_val_log =
-          _rho(current_argument, state) * u_tau * min_wall_dist /
-              (1 / NS::von_karman_constant * std::log(NS::E_turb_constant * y_plus)) -
-          _mu(current_argument, state);
+      // buffer layer
       const auto blending_function = (y_plus - 5.0) / 25.0;
-      const auto wall_val = blending_function * wall_val_log;
-      mu_t_wall = wall_val.value();
+      // the blending depends on the mut_log at y+=30
+      const auto mut_log = mu * (NS::von_karman_constant * 30.0 /
+                                     std::log(std::max(NS::E_turb_constant * 30.0, 1 + 1e-4)) -
+                                 1.0);
+      mu_t = std::max(raw_value(blending_function * mut_log), NS::mu_t_low_limit);
     }
-    // Limiting the minimum turbulent viscosity to 1e-10
-    // The reason is that smaller values can result in over dissipation
-    // when solving the epsilon equation, leading to an instable coupling
-    // in the k-epsilon system
-    // Note: 1e-10 is 10e7 times smaller than the viscosity of water
-    mu_t = std::max(mu_t_wall, 1e-10);
   }
   else
   {
-    // Computing bulk value for turbulent dynamic visocisity
-    const auto time_scale = _k(current_argument, state) / _epsilon(current_argument, state);
+    ADReal time_scale;
+    if (_scale_limiter == "standard")
+    {
+      time_scale =
+          std::max(_k(current_argument, state) / (_epsilon(current_argument, state) + 1e-15),
+                   std::sqrt(nu / (_epsilon(current_argument, state) + 1e-15)));
+    }
+    else
+    {
+      time_scale = _k(current_argument, state) / (_epsilon(current_argument, state) + 1e-15);
+    }
+
     const ADReal mu_t_nl =
         _rho(current_argument, state) * _C_mu * _k(current_argument, state) * time_scale;
     mu_t = mu_t_nl.value();
   }
-
-  return mu_t;
+  // Turbulent viscosity limiter
+  return std::min(mu_t, 1e5 * raw_value(mu));
```

Comment:
Can we make the 1e5 limit a parameter that the user can change? I am good for defaulting this parameter to 1e5

Reply:
Added it